### PR TITLE
Chore: Dynamic fee rate optimisations & Cypress tests

### PR DESCRIPTION
**Pull request by moneymanolis**
- Adjustments to dynamic mode in `fee-selection.html `to account for low and asymmetric fees (very high fee for the next block but very low fees for waiting a bit)
- Cypress tests that use mocked data from the fees fetch API to test different fee environments

### CODE REVIEW

**Review by k9ert — APPROVED** [review]

File: src/cryptoadvance/specter/templates/includes/fee-selection.html [hunk 1]
```diff
@@ -238,11 +238,20 @@ <h4 slot="title">{{ _("Subtract fees from amount") }}</h4>
...
-            let sliderMin = this.fees["minimumFee"]
-            let sliderMax = this.fees["fastestFee"]
-            let average = Math.floor((sliderMin + sliderMax) / 2)
-            this.feesSlider.value = average
+            // For low fee environment
+            if (this.fees["fastestFee"] == 1) {
+                this.feesSlider.max = 1.5
+            }
+            else {
+                this.feesSlider.max = Math.floor(this.fees["fastestFee"] * 1.4)
```
Comment (relativisticelectron) [inline src/cryptoadvance/specter/templates/includes/fee-selection.html:246]: Assume `this.fees["fastestFee"] == 1.1`, then `floor(1.1*1.4) = 1` -->  Stuck slider. I think `ceil` with minimum 2 is the better option.

File: src/cryptoadvance/specter/templates/includes/fee-selection.html [hunk 2]
```diff
@@ -238,11 +238,20 @@ <h4 slot="title">{{ _("Subtract fees from amount") }}</h4>
         // Gets executed after fee initialisation
         initWithFees() {
             // this.feesSlider.min is set above via min_fee
-            this.feesSlider.max = Math.floor(this.fees["fastestFee"] * 1.4)
-            let sliderMin = this.fees["minimumFee"]
-            let sliderMax = this.fees["fastestFee"]
-            let average = Math.floor((sliderMin + sliderMax) / 2)
-            this.feesSlider.value = average
+            // For low fee environment
+            if (this.fees["fastestFee"] == 1) {
```
Comment (relativisticelectron) [inline src/cryptoadvance/specter/templates/includes/fee-selection.html:242]: I triggered this if condition manually, by setting `this.fees["fastestFee"] = 1` just above the if condition. However the slider  maximum is not at  1.5, but at 4.... don't understand why yet....

**Review by relativisticelectron — COMMENTED** [review]
see comments

File: src/cryptoadvance/specter/templates/includes/fee-selection.html (hunk 1 above)
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/includes/fee-selection.html:246]: I'll add that case to the tests and fix it, thanks.

Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/includes/fee-selection.html]: Implemented in https://github.com/cryptoadvance/specter-desktop/pull/1878/commits/1f2228fea88826ed799e937857d6a76bb6bacb55

**Pull request merged**